godot-codegen/src/util.rs

```diff
@@ -241,6 +241,24 @@ pub fn make_enum_definition(enum_: &Enum) -> TokenStream {
     unique_ords.sort();
     unique_ords.dedup();
 
+    let mut derives = vec!["Copy", "Clone", "Eq", "PartialEq", "Debug", "Hash"];
+
+    if enum_.is_bitfield {
+        derives.push("Default");
+    }
+
+    let derives = derives.into_iter().map(ident);
+
+    let index_enum_impl = if let Some(enum_max) = try_count_index_enum(enum_) {
+        quote! {
+            impl crate::obj::IndexEnum for #enum_name {
+                const ENUMERATOR_COUNT: usize = #enum_max;
+            }
+        }
+    } else {
+        TokenStream::new()
+    };
```

Comment:
Good point.

I looked at the documentation for this PR vs the master branch docs at what implements `IndexEnum`, I found that the count stays the same regardless.

I am unsure if a bitfield would ever use this, since, according to the documentation of `IndexEnum`:

> Trait for enums that can be used as indices in arrays.
> 
> The conditions for a Godot enum to be “index-like” are:
> 
>    Contains an enumerator ending in _MAX, which has the highest ordinal (denotes the size).
>    All other enumerators are consecutive integers inside 0..max (no negative ordinals, no gaps).

A bitfield doesn't really work for this. bitfields have gaps (powers of two), and no defined `_MAX`, and don't intrinsically have the possibility of being "index-like" in the first place imo.

So I guess this should be enum only, for those that do satisfy the conditions.

Will make the change if there's no more thoughts on this. :+1:

Reply:
I agree, would probably good to check this for enums only.

Reply:
Wrapped it as an `Option<TokenStream>` to make the intent more clear.

Reply:
Ah, I see. Will do. Followed the original impl of `bitfield_ops`, so assumed I should handle it similarly :+1:

will change ASAP.

I take it as I should change `bitfield_ops` also?

Reply:
Yeah it may not be 100% consistent, but somewhat 😉 

Yes, for the other occurrence too, I mentioned it in a comment.